src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,7 +34,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        
+        String[] addressArray = address.split(", ");
```

Comment:
-having a space explicitly in the split fails to parse addresses without a space, no?
 - e.g.123,clementi ave3,#3-03,S112234
- what would be a better way to handle spaces, think?